src/registry.js

```diff
@@ -0,0 +1,123 @@
+// [VexFlow](http://vexflow.com) - Copyright (c) Mohit Muthanna 2010.
+// @author Mohit Cheppudira
+//
+// ## Description
+//
+// This file implements a registry for VexFlow elements. It allows users
+// to track, query, and manage some subset of generated elements, and
+// dynamically get and set attributes.
+//
+// There are two ways to regiser with a registry:
+//
+// 1) Explicitly call `element.register(registry)`, or,
+// 2) Call `Registry.enableDefaultRegistry(registry)` when ready, and all future
+//    elements will automatically register with it.
+//
+// Once an element is registered, selected attributes are tracked and indexed by
+// the registry. This allows fast look up of elements by attributes like id, type,
+// and class.
+
+export class X extends Error {
```

Comment:
Can we create an Error class factory to avoid the duplication?

```javascript
function makeErrorType(name) {
  return class extends Error {
    constructor() { ... }
  };
}
```